src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -78,25 +78,18 @@ public void lockExclusive(UUID datasetId, String flightId) {
             throw new DatasetLockException("Locking flight id cannot be null");
         }
 
+        logger.info("Lock Operation: Adding exclusive lock for datasetId: {}, flightId: {}", datasetId, flightId);
         // update the dataset entry and lock it by setting the flight id
         String sql = "UPDATE dataset SET flightid = :flightid " +
             "WHERE id = :datasetid AND (flightid IS NULL OR flightid = :flightid) AND CARDINALITY(sharedlock) = 0";
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("datasetid", datasetId)
             .addValue("flightid", flightId);
-        int numRowsUpdated = jdbcTemplate.update(sql, params);
-
-        // if no rows were updated, then throw an exception
-        if (numRowsUpdated == 0) {
-            // this method checks if the dataset exists
-            // if it does not exist, then the method throws a DatasetNotFoundException
-            // we don't need the result (dataset summary) here, just the existence check, so ignore the return value.
-            retrieveSummaryById(datasetId);
-
-            // otherwise, throw a lock exception
-            logger.debug("numRowsUpdated=" + numRowsUpdated);
-            throw new DatasetLockException("Failed to lock the dataset");
-        }
+
+
+        performLockQuery(sql, params, LockType.LockExclusive, datasetId);
+
+        logger.info("Lock Operation: Exclusive lock acquired for dataset {}, flight {}", datasetId, flightId);
```

Comment:
Do you want to leave this logging at info?

Reply:
I think I have a habit of over logging 😃 I've demoted to debug.